sdk/clientcore/core/src/main/java/io/clientcore/core/serialization/json/JsonWriter.java

```diff
@@ -1322,6 +1323,9 @@ public JsonWriter writeUntyped(Object value) throws IOException {
             return writeString(String.valueOf(((Character) value).charValue()));
         } else if (value instanceof JsonSerializable<?>) {
             return ((JsonSerializable<?>) value).toJson(this);
+        } else if (value instanceof BinaryData) {
+            ((BinaryData) value).writeTo(this);
```

Comment:
This is a good add, but I think we might hit a point soon where we need to allow external customization to this as an escape hatch. Ex, if `AnotherType` needs to be supported we should have a way for this to be added without code changes here while we update our code.

Reply:
Discussed offline and decided to add this feature when there is a need to support other new types.